### PR TITLE
Add the Linode provider

### PR DESCRIPTION
**Pull request by btobolaski**
This adds Linode's linode resource. This is a direct port from my [Terraform plugin for Linode][1], I thought I'd finally (since July #1013) get around to creating a pull request for it and see what you have to say about it.

[1]:https://github.com/btobolaski/terraform-provider-linode 

Here are some things that I think should have some additional discussion:
* Changing the linode's size forces a recreate.
  * Linode allows you to resize instances but, its a long process. It can take more than a half hour for the process to complete.
* Doesn't allow you to specify the disk size. I simply allocate 512mb for swap and then use the rest of the instances allotted disk for the main drive.
* Uses the plan's ram in order to identify the size

I'm sure that there are other points that are worthy of discussion but, these were the ones that immediately came to me.

### CODE REVIEW

**Comment by btobolaski** [comment]
Theres probably something obvious that I'm missing but, how do I get the acceptance tests to not run during the travis build?

File: builtin/providers/linode/resource_linode_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/taoh/linodego"
+)
+
+func init() {
+	os.Setenv("TF_ACC", "true")
+}
```
Comment (radeksimko) [inline builtin/providers/linode/resource_linode_test.go]: > Theres probably something obvious that I'm missing but, how do I get the acceptance tests to not run during the travis build?

Just leave out the three lines above. :wink:

**Comment by btobolaski** [comment]
Doh!

**Comment by CpuID** [comment]
Will be nice to see this one merged :)

**Comment by bdwyertech** [comment]
I added some features to this Provider in the following PR.

https://github.com/btobolaski/terraform-provider-linode/pull/1

**Comment by ejaz-ahmed** [comment]
When will it be merged?

**Comment by iMelnik** [comment]
:+1:

**Comment by devinrsmith** [comment]
+1

**Comment by lcacciagioni** [comment]
Any ETA on when this will be merged?

**Comment by stack72** [comment]
Hi @btobolaski 

Apologies for this not being looked at earlier. Please can you have a look at rebasing this PR against master and we can have a try to get it merged?

Thanks

Paul

**Comment by jhezjkp** [comment]
Looking forward to the merge.

**Comment by displague** [comment]
If @btobolaski isn't available, I can take a whack at rebasing if that's alright with the parties involved.

@bdwyertech has an open PR that I would want to include as well at https://github.com/btobolaski/terraform-provider-linode/pull/1

As a developer at Linode, and the maintainer of the vagrant-linode plugin, I'm exciting about having this merged.

**Comment by bdwyertech** [comment]
Glad to see you in on this @displague There is no one better to address the Linode API intricacies and nuances IMHO

**Comment by lcacciagioni** [comment]
Any news on this guys?

**Comment by displague** [comment]
None on this front. This is the first `go` project I would be working on so I expect some learning curve ahead, but so far I haven't had a chance to dive in -- work and vacations and such.

**Comment by displague** [comment]
I've started "something" by looking at the existing plugin and updating that with some of the open PRs and plan-size updates.

When I have that sorted (and reviewed) I'll replay the changes @btobolaski made to turn his awesome plugin into a native support PR.

**Comment by stack72** [comment]
ok, this seems like a dead PR - right now we have some efforts to get this rebased on other branches. I am going to close this out for now. It's been 2 months since i originally posted on this. Looking forward to seeing this reopened when someone has some bandwidth

**Comment by displague** [comment]
Thanks to all the contributions and support from [HashiCorp](https://www.hashicorp.com/), [Linode](https://www.linode.com/), and the community, this provider has become official!

Please **Star** ⭐️ , *Watch* 👁 , and Fork 🍴 :
- https://github.com/terraform-providers/terraform-provider-linode

It is now possible to use `terraform init` to automatically install the Linode Terraform provider without the need to build and install it by hand.

The new documentation is available at:
- https://www.terraform.io/docs/providers/linode/

An example of the provider's features is here:
- https://github.com/terraform-providers/terraform-provider-linode/tree/master/examples

Extra special thanks to @btobolaski whose repo was the base of all this!

--- 

Thanks for reporting this issue @sobrinho -- for getting the ball rolling!

**Pull request closed without merging**